add-controllers

Eugenpak opened the pull request:
Add controllers

Eugenpak committed:
Сохранить ShareItApp

Eugenpak committed:
Сохранить пакет exception

Eugenpak committed:
Сохранить пакет user

Eugenpak committed:
Сохранить изменения в пакете item.dto

Eugenpak committed:
Сохранить изменения в пакете item.model

Eugenpak committed:
Сохранить изменения в пакете item: ItemController

Eugenpak committed:
Сохранить изменения в пакете item: ItemService, ItemServiceImpl

Eugenpak committed:
Сохранить изменения в пакете item: ItemMapper

Eugenpak committed:
Сохранить изменения в пакете item: ItemRepository, ItemRepositoryInMemory

Eugenpak committed:
Checkstyle Spr-14-add-controller-v01

Eugenpak committed:
Checkstyle Spr-14-add-controller-v02

Eugenpak committed:
Checkstyle Spr-14-add-controller-v03

Eugenpak committed:
Checkstyle Spr-14-add-controller-v04

avfyodorov commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,68 @@
+import java.util.List;
 
 /**
  * TODO Sprint add-controllers.
```
Такого рода комментарии лучше удалять. Смысла в них, в общем-то, нет. 🙂
Тем более, что уже done )

Eugenpak replied:
Исправил.

avfyodorov commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,68 @@
+    private final ItemService itemService;
+
+    @GetMapping
+    public List<ItemDto> get(@NotNull @RequestHeader("X-Sharer-User-Id") long userId) {
```
NotNull 

 Обратите внимание, что необходимо будет добавить @validated в контроллер на уровне класса, чтобы проверять параметры метода. В этом случае аннотация @validated устанавливается на уровне класса, даже если она присутствует на методах.

Eugenpak replied:
Исправил.

avfyodorov commented on src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -1,7 +1,20 @@
+import lombok.NoArgsConstructor;
+
 /**
  * TODO Sprint add-controllers.
```
и здесь и дальше. 
Лучше удалить

Eugenpak replied:
Исправил

avfyodorov commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,64 @@
+    }
+
+    @PostMapping
+    public User create(@Valid @RequestBody User user) {
```
Valid @RequestBody User user)

Здесь не должно быть объектов БД. Он же пришёл снаружи, значит должен быть dto.
Передаётся в слой сервиса как dto, там  происходит обработка, а перед там как передать в слой репозитария, делается маппинг в объект базы данных.

Eugenpak replied:
Исправил

avfyodorov commented on src/main/java/ru/practicum/shareit/user/UserController.java:
```diff
@@ -1,12 +1,64 @@
+    }
+
+    @PostMapping
+    public User create(@Valid @RequestBody User user) {
```
public User create(

В этом методе выходной параметр должен  иметь тип UserDto. Вызывается он откуда-то снаружи,  из фронта и передаётся наружу. А фронт,в свою очередь, работает с объектами типа дто, например, UserDto и не имеет доступа  к родным объектам базы данных (User). По крайней мере, не должен иметь. 

Посмотрите, пожалуйста,  аналогичные ситуации и в других методах и в других контроллерах.

Eugenpak replied:
Исправил

avfyodorov reviewed (commented):
Добрый день, Евгений! 

Хорошая работа, от меня всего пара уточнений.

Eugenpak committed:
Исправить ошибки:
https://github.com/Eugenpak/java-shareit/pull/1#discussion_r2062545421
https://github.com/Eugenpak/java-shareit/pull/1#discussion_r2062541804
https://github.com/Eugenpak/java-shareit/pull/1#discussion_r2062529993
https://github.com/Eugenpak/java-shareit/pull/1#discussion_r2062523126

avfyodorov reviewed (commented):
Добрый день, Евгений!

Замечаний нет.
Работа принята.
